fiftyone/core/odm/sample.py

```diff
@@ -438,17 +438,16 @@ def delete_field(cls, field_name):
         delattr(cls, field_name)
 
         # Update dataset meta class
-        # @todo(Tyler) refactor to avoid local import here
-        import fiftyone.core.dataset as fod
-
-        dataset = fod.load_dataset(cls.__name__)
-        dataset._meta.sample_fields = [
-            sf for sf in dataset._meta.sample_fields if sf.name != field_name
+        dataset_doc = DatasetDocument.objects.get(
+            sample_collection_name=cls.__name__
+        )
+        dataset_doc.sample_fields = [
+            sf for sf in dataset_doc.sample_fields if sf.name != field_name
         ]
-        dataset._meta.save()
+        dataset_doc.save()
 
     def _get_repr_fields(self):
-        return ("dataset_name",) + super()._get_repr_fields()
+        return ("collection_name",) + super()._get_repr_fields()
```

Comment:
@tylerganter can we remove `collection_name` since it is now an internal-only concept?